### PR TITLE
SONARPY-1640 S6883: The 12-hour format should be used with the AM/PM marker, otherwise 24-hour format should be used

### PR DESCRIPTION
**Pull request by ghislainpiot**
otherwise 24-hour format should be used,

### CODE REVIEW

**Review by guillaume-dequenne-sonarsource — CHANGES_REQUESTED** [review]
Looks good!
Left some comments to improve on some edge cases and testing in general.

File: python-checks/src/test/resources/checks/strftimeConfusingHourSystem.py [hunk 1]
```diff
@@ -23,3 +23,25 @@ def simple_cases():
...
+        format2 = "%H:%M"
+    t.strftime(format2) # FN because of the limitations of SingleAssignedValue
+
+    # This one is a FN because we don't support f-strings in this rule
+    format19 = f"%I:%M"
+    t.strftime(format19)
+
+    # This one is a FN because we don't support f-strings in this rule
+    format_marker = "%p"
+    t.strftime(f"%I:%M {format_marker}")
```
Comment (guillaume-dequenne-sonarsource) [inline python-checks/src/test/resources/checks/strftimeConfusingHourSystem.py:47]: This one is actually compliant, no?

File: python-checks/src/main/java/org/sonar/python/checks/StrftimeConfusingHourSystemCheck.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        effectiveMessage = MESSAGE_12_HOURS;
+      }
+      if (effectiveMessage != null) {
+        var issue = context.addIssue(primaryLocation, effectiveMessage);
+        if (primaryLocation != expression) {
+          issue.secondary(stringLiteral, MESSAGE_SECONDARY_LOCATION);
+        }
+      }
+    }
+  }
```
Comment (maksim-grebeniuk-sonarsource) [inline python-checks/src/main/java/org/sonar/python/checks/StrftimeConfusingHourSystemCheck.java:74]: This could be implemented in the following way:
```java
  private static void checkExpression(SubscriptionContext context, Expression expression, Tree primaryLocation) {
    Expressions.ifNameGetSingleAssignedNonNameValue(expression)
      .filter(StringLiteral.class::isInstance)
      .map(StringLiteral.class::cast)
      .filter(stringLiteral -> stringLiteral.stringElements().stream().noneMatch(stringElement -> "f".equalsIgnoreCase(stringElement.prefix())))
      .ifPresent(stringLiteral -> checkDateFormatStringLiteral(context, primaryLocation, stringLiteral));
  }

  private static void checkDateFormatStringLiteral(SubscriptionContext context, Tree primaryLocation, StringLiteral stringLiteral) {
    String value = stringLiteral.trimmedQuotesValue();
    String effectiveMessage = null;
    if (value.contains("%H") && value.contains("%p")) {
      effectiveMessage = MESSAGE;
    } else if (value.contains("%I") && !value.contains("%p")) {
      effectiveMessage = MESSAGE_12_HOURS;
    }
    if (effectiveMessage != null) {
      var issue = context.addIssue(primaryLocation, effectiveMessage);
      if (primaryLocation != stringLiteral) {
        issue.secondary(stringLiteral, MESSAGE_SECONDARY_LOCATION);
      }
    }
  }
```

**Review by maksim-grebeniuk-sonarsource — APPROVED** [review]
LGTM, see one comment, but it is minor and up to you to accept this changes
